wrapper.go

```diff
@@ -1,121 +1,197 @@
 package log
 
+import (
+	"sync"
+)
+
 // A global variable so that l functions can be directly accessed.
-var l Logger
+var (
+	l  Logger = Noop{}
+	mu sync.RWMutex
+)
 
 // NewLogger returns an instance of logger.
+// Deprecated: prefer SetGlobalLogger
 func NewLogger(logger Logger) {
+	mu.Lock()
+	defer mu.Unlock()
+	l = logger
+}
+
+func SetGlobalLogger(logger Logger) {
```

Comment:
This could be only SetLogger

Reply:
I choose this name because while `NewLogger` functions produce any number of `Logger` instances, the `SetGlobalLogger` deal if one `Logger` instance in specific, the global one. Hence I believe to be better to make explicitly the responsibility of this function. This is a common design, like [zap's `ReplaceGlobals` functions](https://pkg.go.dev/go.uber.org/zap#ReplaceGlobals).